helix-term/src/commands/engine/steel.rs

```diff
@@ -692,6 +692,12 @@ fn load_configuration_api(engine: &mut Engine, generate_sources: bool) {
 fn load_editor_api(engine: &mut Engine, generate_sources: bool) {
     let mut module = BuiltInModule::new("helix/core/editor");
 
+    // Types
+    module.register_fn("Action::Load", || Action::Load);
+    module.register_fn("Action::Replace", || Action::Replace);
+    module.register_fn("Action::HorizontalSplit", || Action::HorizontalSplit);
+    module.register_fn("Action::VerticalSplit", || Action::VerticalSplit);
```

Comment:
Yep - sorry I didn't get back faster, this works. My only nit would be that we try to align with scheme naming conventions, so do something like:

`action/load` or `action-load` - that way it doesn't look too stylistically out of place

Reply:
It's all right, new things stick better when I am finding them out instead of being told the solution 😅 

I am fine with both, maybe `action/load` conveys better the fact that it's an enum, although I haven't seen this syntax much elsewhere... So which one do you suggest?

Reply:
I've been using the `<type>/<variant>` pattern while defining the component api, so we should go with that style

Reply:
e.g.

```rust
        .register_value("Color/Reset", Color::Reset.into_steelval().unwrap())
        .register_value("Color/Black", Color::Black.into_steelval().unwrap())
        .register_value("Color/Red", Color::Red.into_steelval().unwrap())
        .register_value("Color/White", Color::White.into_steelval().unwrap())
        .register_value("Color/Green", Color::Green.into_steelval().unwrap())
        .register_value("Color/Yellow", Color::Yellow.into_steelval().unwrap())
        .register_value("Color/Blue", Color::Blue.into_steelval().unwrap())
        .register_value("Color/Magenta", Color::Magenta.into_steelval().unwrap())
        .register_value("Color/Cyan", Color::Cyan.into_steelval().unwrap())
        .register_value("Color/Gray", Color::Gray.into_steelval().unwrap())
        .register_value("Color/LightRed", Color::LightRed.into_steelval().unwrap())
        .register_value(
            "Color/LightGreen",
            Color::LightGreen.into_steelval().unwrap(),
        )
        .register_value(
            "Color/LightYellow",
            Color::LightYellow.into_steelval().unwrap(),
        )
        .register_value("Color/LightBlue", Color::LightBlue.into_steelval().unwrap())
        .register_value(
            "Color/LightMagenta",
            Color::LightMagenta.into_steelval().unwrap(),
        )
        .register_value("Color/LightCyan", Color::LightCyan.into_steelval().unwrap())
        .register_value("Color/LightGray", Color::LightGray.into_steelval().unwrap())
        .register_fn("Color/rgb", Color::Rgb)
        .register_fn("Color/Indexed", Color::Indexed)
        .register_fn("set-style-fg!", |style: &mut Style, color: Color| {
            style.fg = Some(color);
        })
        .register_fn("style-fg", Style::fg)
        .register_fn("style-bg", Style::bg)
        .register_fn("set-style-bg!", |style: &mut Style, color: Color| {
            style.bg = Some(color);
        })
        .register_fn("style-underline-color", Style::underline_color)
        .register_fn("style-underline-style", Style::underline_style)
        .register_value(
            "Underline/Reset",
            UnderlineStyle::Reset.into_steelval().unwrap(),
        )
        .register_value(
            "Underline/Line",
            UnderlineStyle::Line.into_steelval().unwrap(),
        )
        .register_value(
            "Underline/Curl",
            UnderlineStyle::Curl.into_steelval().unwrap(),
        )
        .register_value(
            "Underline/Dotted",
            UnderlineStyle::Dotted.into_steelval().unwrap(),
        )
        .register_value(
            "Underline/Dashed",
            UnderlineStyle::Dashed.into_steelval().unwrap(),
        )
        .register_value(
            "Underline/DoubleLine",
            UnderlineStyle::DoubleLine.into_steelval().unwrap(),
        )
        .register_fn("style", || Style::default())
        .register_value(
            "event-result/consume",
            SteelEventResult::Consumed.into_steelval().unwrap(),
        )
        .register_value(
            "event-result/consume-without-rerender",
            SteelEventResult::ConsumedWithoutRerender
                .into_steelval()
                .unwrap(),
        )
        .register_value(
            "event-result/ignore",
            SteelEventResult::Ignored.into_steelval().unwrap(),
        )
        .register_value(
            "event-result/close",
            SteelEventResult::Close.into_steelval().unwrap(),
        )
```

Reply:
Ah, sorry I haven't came across this peice of code yet. Updated in 4f1d35dfbe02

Reply:
no problem! lgtm